tests/kolainst/destructive/container-image

```diff
@@ -0,0 +1,83 @@
+#!/bin/bash
+#
+# Copyright (C) 2021 Red Hat, Inc.
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+# Boston, MA 02111-1307, USA.
+
+set -euo pipefail
+
+. ${KOLA_EXT_DATA}/libtest.sh
+
+set -x
+cd $(mktemp -d)
+
+case "${AUTOPKGTEST_REBOOT_MARK:-}" in
+  "")
+    # Test rebase
+    checksum=$(rpm-ostree status --json | jq -r '.deployments[0].checksum')
+    rpm-ostree ex-container export --repo=/ostree/repo ${checksum} containers-storage:localhost/fcos
+    rpm-ostree rebase containers-storage:localhost/fcos:latest --experimental
+    rpmostree_assert_status ".deployments[0][\"container-image-reference\"] == \"containers-storage:localhost/fcos:latest\""
+    rpmostree_assert_status ".deployments[0][\"checksum\"] == \"${checksum}\""
+    echo "ok rebase to container image reference"
+
+    /tmp/autopkgtest-reboot 1
+    ;;
+  1)
+    if rpm-ostree deploy 42 2>err.txt; then
+        fatal "unexpectedly deployed version from container image reference"
+    fi
+    assert_file_has_content err.txt 'Cannot look up version while tracking a container image reference'
+    echo "ok cannot deploy when tracking container image"
+
+    # Test layering
+    # Disable repos, no Internet access should be required
+    rm -rf /etc/yum.repos.d/
+    if rpm -q foo 2>/dev/null; then
+      fatal "found foo"
+    fi
+    rpm-ostree install ${KOLA_EXT_DATA}/rpm-repos/0/packages/x86_64/foo-1.2-3.x86_64.rpm
+    echo "ok install foo locally"
+
+    /tmp/autopkgtest-reboot 2
```

Comment:
This is fine as is, but I think you can save a reboot by committing the pending deployment instead of rebooting and having it be the booted.  (Something to look at after merge in other words)

Reply:
ah true. I was also thinking we wanted to do a `rpm -q foo` after the reboot to make sure the package is really there. Anyways, probably smart to save a reboot since I did notice that this test was taking a bit longer than the others :)